compiler/rustc_typeck/src/coherence/orphan.rs

```diff
@@ -244,10 +244,17 @@ fn emit_orphan_check_error<'tcx>(
                 };
                 let this = "this".to_string();
                 let (ty, postfix) = match &ty.kind() {
-                    ty::Slice(_) => (this, " because slices are always foreign"),
-                    ty::Array(..) => (this, " because arrays are always foreign"),
-                    ty::Tuple(..) => (this, " because tuples are always foreign"),
-                    _ => (format!("`{}`", ty), ""),
+                    ty::Slice(_) => (this, " because slices are always foreign".to_string()),
+                    ty::Array(..) => (this, " because arrays are always foreign".to_string()),
+                    ty::Tuple(..) => (this, " because tuples are always foreign".to_string()),
+                    ty::RawPtr(ptr_ty) => {
+                        let sugg = format!(
+                            ", you may want implement it on `{}` or `&{}`",
```

Comment:
```suggestion
                            ", you may want implement it on `{}` or `&mut {}`",
```
Mh! maybe in this case maybe we need to check also if we need to add the `mut` keyword?